### PR TITLE
build: Fix / improve coverage scripts

### PR DESCRIPTION
**Pull request by hebasto**
The first commit ensures early error catching.

The second commit adds the `-g` flag that was missed during the migration from Autotools.

This PR is intended to be tested with GCC compiler (as clang support is still under [scrutiny](https://github.com/hebasto/bitcoin/issues/341)). Depending on the `lcov` version, additional flags `-DCMAKE_C_FLAGS="-fprofile-update=atomic" -DCMAKE_CXX_FLAGS="-fprofile-update=atomic"` may be required.

### CODE REVIEW

**Review by tdb3 — APPROVED** [review]
ACK 6e5254cf687578ebb686628cf89f2509d415f04f

Light CR and test.

```
cmake -B build -DENABLE_WALLET=OFF -DCMAKE_BUILD_TYPE=Coverage
cmake --build build -j18
cmake -P build/Coverage.cmake
```

Induced failure (and received error as expected) with:
```diff
 execute_process(
+  COMMAND rm test_bitcoin.info
   COMMAND ${LCOV_FILTER_COMMAND} test_bitcoin.info test_bitcoin_filtered.info
   WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
   COMMAND_ERROR_IS_FATAL ANY
```

File: cmake/script/CoverageFuzz.cmake [hunk 1]
```diff
@@ -16,27 +16,33 @@ execute_process(
 execute_process(
```
[the remark this answers is not in the document]
Comment (hebasto) [reply cmake/script/CoverageFuzz.cmake:16]: You mean for the `lcov` command?

[the remark this answers is not in the document]
Comment (hebasto) [reply cmake/script/CoverageFuzz.cmake:21]: Thanks! Added.

**Comment by hebasto** [comment]
Rebased to resolve a conflict with the merged https://github.com/bitcoin/bitcoin/pull/30804.

**Review by tdb3 — APPROVED** [review]
cr re ACK d9fcbfc3727e06e6f57d4ab09861f3212d558426

**Pull request merged**